crates/astria-sequencer/src/metrics_init.rs

```diff
@@ -0,0 +1,173 @@
+//! Crate-specific metrics functionality.
+//!
+//! Registers metrics & lists constants to be used as metric names throughout crate.
+
+use metrics::{
+    describe_counter,
+    describe_gauge,
+    describe_histogram,
+    Unit,
+};
+
+/// Registers all metrics used by this crate.
+pub fn register() {
+    describe_counter!(
+        PREPARE_PROPOSAL_EXCLUDED_TRANSACTIONS_DECODE_FAILURE,
+        Unit::Count,
+        "The number of transactions that have been excluded from blocks due to failing to decode"
+    );
+
+    describe_counter!(
+        PREPARE_PROPOSAL_EXCLUDED_TRANSACTIONS_COMETBFT_SPACE,
+        Unit::Count,
+        "The number of transactions that have been excluded from blocks due to running out of \
+         space in the cometbft block"
+    );
+
+    describe_counter!(
+        PREPARE_PROPOSAL_EXCLUDED_TRANSACTIONS_SEQUENCER_SPACE,
+        Unit::Count,
+        "The number of transactions that have been excluded from blocks due to running out of \
+         space in the sequencer block"
+    );
+
+    describe_counter!(
+        PREPARE_PROPOSAL_EXCLUDED_TRANSACTIONS_FAILED_EXECUTION,
+        Unit::Count,
+        "The number of transactions that have been excluded from blocks due to failing to execute"
+    );
+
+    describe_gauge!(
+        PREPARE_PROPOSAL_EXCLUDED_TRANSACTIONS,
+        Unit::Count,
+        "The number of excluded transactions in a proposal being prepared"
+    );
+
+    describe_gauge!(
+        PROCESS_PROPOSAL_DEPOSIT_TRANSACTIONS,
+        Unit::Count,
+        "The number of deposit transactions in a proposal being processed"
+    );
+
+    describe_counter!(
+        PROCESS_PROPOSAL_SKIPPED_PROPOSAL,
+        Unit::Count,
+        "The number of times our submitted prepared proposal was skipped in process proposal"
+    );
+
+    describe_counter!(
+        CHECK_TX_REMOVED_TOO_LARGE,
+        Unit::Count,
+        "The number of transactions that have been removed from the mempool due to being too large"
+    );
+
+    describe_counter!(
+        CHECK_TX_REMOVED_FAILED_TO_DECODE_PROTOBUF,
+        Unit::Count,
+        "The number of transactions that have been removed from the mempool due to failing to \
+         decode from bytes into a protobuf"
+    );
+
+    describe_counter!(
+        CHECK_TX_REMOVED_FAILED_TO_DECODE_SIGNED_TRANSACTION,
+        Unit::Count,
+        "The number of transactions that have been removed from the mempool due to failing to \
+         decode from a protobuf into a valid signed transaction"
+    );
```

Comment:
what use cases do you see for these? people can submit arbitrary invalid bytes which would increase these

Reply:
I added them for completeness with the other counters. The nonce and balance ones are fun to watch go up. I can remove them if you don't think they're needed

Reply:
removed them!